controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +183,71 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	}
+
+	if !r.storageConsumer.Spec.Enable {
```

Comment:
This decision should happen outside of this function.
The entire reconcile flow need to be separated into 2 cases, Enabled vs Not-Enabled. 
This is important, if not you will find yourself adding an `if Enabled` and `if !Enabled` all across the reconcile function.